From: Alexander Klepikov <klepikov_a@up.ua>
Date: Fri, 2 Jun 2017 11:21:27 +0200
Subject: [PATCH] add support for the Framed-MTU Radius attribute

http://ppp.samba.org/cgi-bin/ppp-bugs/incoming?id=1532

This patch allows radius plugin to deal with Framed-MTU Radius attribute and to
set MTU on interface.
---
 pppd/plugins/radius/radius.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/pppd/plugins/radius/radius.c b/pppd/plugins/radius/radius.c
index 4ba5f523ea07..06e00590b635 100644
--- a/pppd/plugins/radius/radius.c
+++ b/pppd/plugins/radius/radius.c
@@ -651,6 +651,9 @@ radius_setparams(VALUE_PAIR *vp, char *msg, REQUEST_INFO *req_info,
 		    memcpy(rstate.class, vp->strvalue, rstate.class_len);
 		} /* else too big for our buffer - ignore it */
 		break;
+	    case PW_FRAMED_MTU:
+		netif_set_mtu(rstate.client_port,MIN(netif_get_mtu(rstate.client_port),vp->lvalue));
+		break;
 	    }
 
 
